src/NHibernate.Test/NHSpecificTest/GH1226/Fixture.cs

```diff
@@ -0,0 +1,129 @@
+﻿using System.Linq;
+using NHibernate.Engine;
+using NHibernate.Persister.Entity;
+using NUnit.Framework;
+
+namespace NHibernate.Test.NHSpecificTest.GH1226
+{
+	[TestFixture]
+	public class Fixture : BugTestCase
+	{
+		protected override void OnSetUp()
+		{
+			base.OnSetUp();
+
+			using (var session = OpenSession())
+			{
+				using (var tx = session.BeginTransaction())
+				{
+					var bank = new Bank { Code = "01234" };
+					session.Save(bank);
+
+					var account = new Account { Bank = bank };
+					session.Save(account);
+
+					var account2 = new Account { Bank = bank };
+					session.Save(account2);
+
+					tx.Commit();
+				}
+			}
+			Sfi.Statistics.IsStatisticsEnabled = true;
+		}
+
+		[Test]
+		public void BankShouldBeJoinFetched()
+		{
+			using (var session = OpenSession())
+			using (var tx = session.BeginTransaction())
+			{
+				// Bug only occurs if the Banks are already in the session cache.
```

Comment:
Since the old code has changed, could this case please be checked as well?

Reply:
You mean, checking eager fetching works for `Bank` not already in session too? It is likely this is already tested somewhere else, but why not.

Reply:
Done.